org.eclipse.dartboard/src/org/eclipse/dartboard/project/DartFileWizard.java

```diff
@@ -0,0 +1,125 @@
+/*******************************************************************************
+ * Copyright (c) 2019 vogella GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Lakshminarayana Nekkanti 
+ *******************************************************************************/
+package org.eclipse.dartboard.project;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.InvocationTargetException;
+
+import org.eclipse.core.resources.IContainer;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IWorkspaceRoot;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.dartboard.Messages;
+import org.eclipse.dartboard.util.StatusUtil;
+import org.eclipse.jface.operation.IRunnableWithProgress;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.jface.wizard.Wizard;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.ui.INewWizard;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.ide.IDE;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DartFileWizard extends Wizard implements INewWizard {
+	private static final Logger LOG = LoggerFactory.getLogger(DartFileWizard.class);
+
+	private DartFilePage dartFilePage;
+	private IStructuredSelection selection;
+
+	@Override
+	public void init(IWorkbench workbench, IStructuredSelection selection) {
+		this.selection = selection;
+		setWindowTitle(Messages.NewFile_windowTitle);
+		setNeedsProgressMonitor(true);
+	}
+
+	@Override
+	public void addPages() {
+		dartFilePage = new DartFilePage(DartFilePage.class.getSimpleName(), selection);
+		dartFilePage.setTitle(Messages.NewFile_title);
+		dartFilePage.setDescription(Messages.NewFile_description);
+		dartFilePage.setFileExtension("dart"); //$NON-NLS-1$
+		dartFilePage.setFileName("NewDart"); //$NON-NLS-1$

```

Comment:
The default name has a few problems:
- It's not compliant with the Dart style guide.
- It doesn't seem to be a good default.

As I can't think of a good default file name (as you rarely create files with default names), I'am suggesting to remove the default name.